src/api/Feed.js

```diff
@@ -615,18 +646,55 @@ class Feed extends Base {
             completion_rule: completionRule,
         };
 
-        this.tasksNewAPI = new TasksNewAPI(this.options);
-        this.tasksNewAPI.createTask({
-            file,
-            task: taskPayload,
-            successCallback: (taskData: Task) => {
-                this.addPendingItem(this.file.id, currentUser, pendingTask);
-                this.createTaskNewSuccessCallback(file, uuid, taskData, assignees, successCallback, errorCallback);
-            },
-            errorCallback: (e: ElementsXhrError, code: string) => {
-                this.feedErrorCallback(false, e, code);
-            },
-        });
+        // create request for the size of each group by ID
+        const groupInfoPromises: Array<Promise<any>> = assignees
```

Comment:
this new block of code here looks almost the same as the one above added to `updateTaskNew`.. is this intentional? are there opportunities to reuse?

Reply:
in the other block, i added a TODO for reconciling these two: this one uses promise pattern, while the other uses asyn/await. I didn't want to confuse this review with also converting one of them to the other format, since that would likely change additional code in non-functional ways.

Reply:
👍